diffrax/_step_size_controller/constant.py

```diff
@@ -47,15 +52,40 @@ def adapt_step_size(
         args: Args,
         y_error: Y | None,
         error_order: RealScalarLike | None,
-        controller_state: RealScalarLike,
-    ) -> tuple[bool, RealScalarLike, RealScalarLike, bool, RealScalarLike, RESULTS]:
+        controller_state: tuple[IntScalarLike, IntScalarLike, RealScalarLike, RealScalarLike]
+    ) -> tuple[bool, 
+            RealScalarLike,
+            RealScalarLike,
+            bool,
+            tuple[IntScalarLike, IntScalarLike, RealScalarLike, RealScalarLike], 
+            RESULTS
+        ]:
         del t0, y0, y1_candidate, args, y_error, error_order
+        step, num_steps, t0_sim, t1_sim = controller_state
+        step += 1
+        eqx.error_if(step,
```

Comment:
Note that this check won't actually do anything because its return value is unused.

That aside, I don't think we should have this - these checks add a lot of runtime overhead so I try to minimise their use.